api/src/main/java/javax/enterprise/inject/spi/container/Container.java

```diff
@@ -0,0 +1,341 @@
+     *
+     * @param decoratorClasses classes of the decorators to enable.
+     * @return self
+     */
+    Container decorators(Class<?>... decoratorClasses);
+
+    /**
+     * Add a decorator class to the list of enabled decorators for a synthetic bean archive.
+     *
+     * @param decoratorClass class of the decorator to add.
+     * @return self
+     */
+    Container addDecorator(Class<?> decoratorClass);
+
+    /**
+     * Select alternatives for a synthetic bean archive.
+     *
+     * @param alternativeClasses classes of the alternatives to select
+     * @return self
+     */
+    Container alternatives(Class<?>... alternativeClasses);
+
+    /**
+     * Add an alternative class to the list of selected alternatives for a synthetic bean archive.
+     *
+     * @param alternativeClass class of the alternatives to add
+     * @return self
+     */
+    Container addAlternative(Class<?> alternativeClass);
+
+    /**
+     * Select alternative stereotypes for a synthetic bean archive.
+     *
+     * @param alternativeStereotypeClasses alternatives stereotypes to select
+     * @return self
+     */
+    Container alternativeStereotypes(Class<? extends Annotation>... alternativeStereotypeClasses);
+
+    /**
+     * Add an alternative stereotype class to the list of selected alternative stereotypes for a synthetic bean archive.
+     *
+     * @param alternativeStereotypeClass alternative stereotype to add
+     * @return self
+     */
+    Container addAlternativeStereotype(Class<? extends Annotation> alternativeStereotypeClass);
+
+    /**
+     * Set the configuration property for the container
+     *
+     * @param key property name
+     * @param value property value
+     * @return self
+     */
+    Container property(String key, Object value);
+
+    /**
+     * Set all the configuration properties.
+     *
+     * @param properties a map containing properties to add
+     * @return self
+     */
+    Container properties(Map<String, Object> properties);
+
+    /**
+     * Returns a {@link BeanConfigurator} intialized with the provided type
+     * to help you define an additional bean to add to the deployment.
+     *
+     * Bean will be created when initialize is called
+     *
+     * @param type type of the bean
+     * @return a BeanConfigurator ro define a custom bean
+     */
+    <T> BeanConfigurator<T> configureBean(Class<T> type);
+
+
+    /**
+     * Add the provided beans to set of discovered beans.
+     *
+     * @param beans beans to add
+     * @return self
+     */
+    Container AddBeans(Bean<?>... beans);
+
+    /**
+     *
+     * Add the provided bean to set of discovered beans.
+     *
+     * @param bean bean to add
+     * @return self
+     */
+    Container addBean(Bean<?> bean);
+
+
+    /**
+     * @return self
+     * @see #disableDiscovery()
+     */
+    Container enableDiscovery();
```

Comment:
I agree.  One thought I had, which may make this more usable:

`CDI<Object> container = ContainerProvider.getContainer().initialize()`

Gives you back a CDI container that has everything already found from the classpath.  Whereas

`CDI<Object> container = ContainerProvider.getContainer().addPackages(true, "some.base.package").initialize()`

Gives you a CDI instance that only refers to the beans contained within those classes (assuming we're following the rules of what is a managed bean)